### PR TITLE
Add `UpgradeView` and `Upgrade` component

### PR DESCRIPTION
**Pull request by veado**
- [x] Add `upgrade` route (incl. tests)
- [x] Add `UpgradeView`
- [x] Add `Upgrade` component
- [x] Add `MaxBalanceButton` component + story
- [x] Add `underline` button
- [x] Quick fix background of `Input` in dark mode

Part of #783

### CODE REVIEW

File: src/renderer/components/wallet/txs/upgrade/Upgrade.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    upgrade$: (_: SendTxParams): UpgradeRuneTxState$ =>
+      mockTxState$([
+        {
+          steps: { current: 1, total },
+          status: RD.failure({ errorId: ErrorId.VALIDATE_POOL, msg: 'invalid pool address' })
+        }
+      ])
+  }
+  return <Upgrade {...props} />
+}
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/wallet/txs/upgrade/Upgrade.stories.tsx:87]: for some reason i can not see any results there
image

File: src/renderer/components/wallet/txs/TxForm.style.tsx [hunk 2]
```diff
@@ -72,3 +72,38 @@ export const ButtonLinkIcon = styled(SelectOutlined)`
 export const Text = styled('span')`
   color: ${palette('text', 2)};
 `
+
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/wallet/txs/TxForm.style.tsx:75]: could you pls add additional styles for this form to "stretch" it for the whole screen ? (all needed style changes are displayed at the screenshot)
image

File: src/renderer/components/wallet/txs/TxForm.style.tsx [hunk 3]
```diff
@@ -72,3 +72,38 @@ export const ButtonLinkIcon = styled(SelectOutlined)`
...
+export const SuccessExtraContainer = styled.div`
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/wallet/txs/TxForm.style.tsx:76]: also make a container to have min 100% of possible height and align by vertical at the center

File: src/renderer/components/wallet/txs/upgrade/Upgrade.tsx [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+  useEffect(() => {
+    // Whenever `amountToUpgrade` has been updated, we put it back into input field
+    form.setFieldsValue({
+      amount: baseToAsset(amountToUpgrade).amount()
+    })
+  }, [amountToUpgrade, form])
+
+  const amountValidator = useCallback(
+    async (_: unknown, value: BigNumber) => {
+      console.log('validator:', value.toString())
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/wallet/txs/upgrade/Upgrade.tsx:160]: [nitpick]

File: src/renderer/components/wallet/txs/upgrade/Upgrade.tsx [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+          msg1: intl.formatMessage({ id: 'wallet.errors.amount.shouldBeNumber' }),
+          msg2: intl.formatMessage({ id: 'wallet.errors.amount.shouldBeGreaterThan' }, { amount: '0' }),
+          msg3: intl.formatMessage({ id: 'wallet.errors.amount.shouldBeLessThanBalance' })
+        }
+      })
+    },
+    [intl, maxAmount]
+  )
+
+  const onChangeInput = useCallback(
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/components/wallet/txs/upgrade/Upgrade.tsx:174]: pls merge latest master. we can not input decimal values right now. i remember there was a fix for that

File: src/renderer/services/wallet/types.ts [hunk 6]
```diff
@@ -122,3 +122,21 @@ export type LedgerTxHashLD = LiveData<LedgerApiError, string>
...
 export type LedgerAddressRD = RD.RemoteData<LedgerErrorId, Address>
 export type LedgerAddressLD = LiveData<LedgerErrorId, Address>
+
+/**
+ * State to reflect status of upgrading Rune
+ *
+ * Two steps are needed:
+ * 1. Health check (pool address)
+ * 1. Send tx
+ * 2. Check status of tx
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/wallet/types.ts:132]: [nitpick]
```
* Three steps are needed:
* 1. Health check (pool address)
* 2. Send tx
* 3. Check status of tx
 ```

File: src/renderer/services/wallet/types.ts [hunk 7]
```diff
@@ -122,3 +122,21 @@ export type LedgerTxHashLD = LiveData<LedgerApiError, string>
...
+ *
+ */
+export type UpgradeRuneTxState = {
+  // State of steps (current step + total number of steps)
+  readonly steps: { current: number; readonly total: number }
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/wallet/types.ts:137]: just to check in terms of typings maybe it's better to use
```ts
readonly total: 3 
```
?

**Review by thatStrangeGuyThorchain — CHANGES_REQUESTED** [review]

File: src/renderer/components/wallet/txs/upgrade/Upgrade.stories.tsx (hunk 1 above)
Comment (veado) [reply src/renderer/components/wallet/txs/upgrade/Upgrade.stories.tsx:87]: Click `Upgrade` button, enter mocked password (`123`) and then you will see the error screen.

File: src/renderer/components/wallet/txs/TxForm.style.tsx (hunk 3 above)
Comment (veado) [reply src/renderer/components/wallet/txs/TxForm.style.tsx:76]: Code pls ;)

File: src/renderer/components/wallet/txs/TxForm.style.tsx (hunk 2 above)
Comment (veado) [reply src/renderer/components/wallet/txs/TxForm.style.tsx:75]: We don't do it in other forms (see send forms), let's keep the same here.

File: src/renderer/components/wallet/txs/upgrade/Upgrade.tsx (hunk 5 above)
Comment (veado) [reply src/renderer/components/wallet/txs/upgrade/Upgrade.tsx:174]: Ok

File: src/renderer/components/wallet/txs/TxForm.style.tsx (hunk 2 above)
Comment (thatStrangeGuyThorchain) [reply src/renderer/components/wallet/txs/TxForm.style.tsx:75]: ok. but we need to fix them all in future to match figma-designs
image

File: src/renderer/components/wallet/txs/TxForm.style.tsx (hunk 3 above)
Comment (thatStrangeGuyThorchain) [reply src/renderer/components/wallet/txs/TxForm.style.tsx:76]: forget this as we desided to keep send forms similar

File: src/renderer/services/wallet/types.ts [hunk 8]
```diff
@@ -122,3 +122,22 @@ export type LedgerTxHashLD = LiveData<LedgerApiError, string>
...
 export type LedgerAddressRD = RD.RemoteData<LedgerErrorId, Address>
 export type LedgerAddressLD = LiveData<LedgerErrorId, Address>
+
+/**
+ * State to reflect status of upgrading Rune
+ *
+ * Three steps are needed:
+ * 1. Health check (pool address)
+ * 1. Send tx
+ * 2. Check status of tx
```
Comment (thatStrangeGuyThorchain) [inline src/renderer/services/wallet/types.ts:132]: [nitpick] still a typo :sweat_smile: but nevermind right now

**Review by thatStrangeGuyThorchain — DISMISSED** [review]

File: src/renderer/components/wallet/txs/TxForm.style.tsx (hunk 3 above)
Comment (veado) [reply src/renderer/components/wallet/txs/TxForm.style.tsx:76]: I've updated the styles with latest commit c15b57d - pls check it again.

**Review by thatStrangeGuyThorchain — APPROVED** [review]

**Pull request merged**